sphere.cpp

```diff
@@ -0,0 +1,50 @@
+#include "sphere.h"
+
+Sphere::Sphere() {
+  for (Element element : ELEMENTLIST) {
+    insert({element, 0});
+  }
+}
+
+Sphere::Sphere(std::map<Element, int> gems) {
+  for (Element element : ELEMENTLIST) {
+    if (gems.find(element) != gems.end())
```

Comment:
I think you can replace this with:
```suggestion
    if (gems.contains(element))
```

Reply:
Note this needs C++20, we may need to change our configuration in Bazel to use it.